rules/prefer-reflect-apply.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+const astUtils = require('eslint-ast-utils');
+const getDocsUrl = require('./utils/get-docs-url');
+
+const isApplySignature = (argument1, argument2) => (
+	((argument1.type === 'Literal' && argument1.raw === 'null') ||
+		argument1.type === 'ThisExpression') &&
+	(argument2.type === 'ArrayExpression' ||
+		(argument2.type === 'Identifier' &&
+		argument2.name === 'arguments'))
+);
+
+const getReflectApplyCall = (sourceCode, func, receiver, args) => (
+	`Reflect.apply(${sourceCode.getText(func)}, ${sourceCode.getText(receiver)}, ${sourceCode.getText(args)})`
+);
+
+const fixDirectApplyCall = (node, sourceCode) => {
+	if (
+		astUtils.getPropertyName(node.callee) === 'apply' &&
+		node.arguments.length === 2 &&
+		isApplySignature(node.arguments[0], node.arguments[1])
+	) {
+		return fixer => (
+			fixer.replaceText(
+				node,
+				getReflectApplyCall(sourceCode, node.callee.object, node.arguments[0], node.arguments[1])
+			)
+		);
+	}
+
+	return null;
+};
+
+const fixFunctionPrototypeCall = (node, sourceCode) => {
+	if (
+		astUtils.getPropertyName(node.callee) === 'call' &&
+		sourceCode.getText(node.callee.object) === 'Function.prototype.apply' &&
```

Comment:
I think it would be better to check the method chain in the actual AST, than comparing string representation of the resulting code.